napari/__main__.py

```diff
@@ -234,6 +246,18 @@ def _run():
         runpy.run_path(args.paths[0])
 
     else:
+        if args.with_:
```

Comment:
@tlambert03 do these if clauses need to be separate? Is it just for the fast failing? I'm not sure if the usability win is worth the code duplication tbh... @sofroniewn any opinion?

Reply:
Would be nice to avoid, but could come in a follow-up refactor, I havn't looked at details to see how easy/ hard that would be

Reply:
Yeah, the alternative is that the splash screen shows, it takes a little longer, it looks like it’s gonna load, and then it fails.  I figured the end user experience was worth the code dupe, but can undo it if you guys want.  Don’t feel too strongly either way

Reply:
Let's leave duplicate code for now then, that sounds less desirable